InventoryComponents/InventoryConsumablesComponent.gd

```diff
@@ -27,37 +29,33 @@ func remove_consumable_at_place(pos : int) -> void:
 	pass
 
 func consume_consumable_at_place(pos : int) -> void:
-	prepare_consumable_array()
-	
 	if(held_consumables[pos] == null):
 		return
 	
 	held_consumables[pos].on_consume()
 	remove_consumable_at_place(pos)
 	pass
 
-func prepare_consumable_array() -> void:
-	while(held_consumables.size() < consumables_limit):
+func _update_consumable_limit(new_amount : int = max_consumable_number) -> void:
+	while(held_consumables.size() < new_amount):
 		held_consumables.append(null)
 	
-	while(held_consumables.size() > consumables_limit):
+	while(held_consumables.size() > new_amount):
 		held_consumables.pop_back()
```

Comment:
I'll change this so that the player is given the item if it were to be in that slot

Reply:
i would rather change it so that you just delete one null and the item stays in the same relative position in the list (ie at the end), and if not possible then the player loses an item

Reply:
oh given, i saw the commit, i saw what you mean